Make update and create conform to json-api resource structure

bobholt opened the pull request:
- move request and response data under "data" key instead of type key
- parse update and create responses through json-formatter
- update unit tests for create and update to use real models
- add integration test for update because i needed it to find a bug in the
unit tests

fixes #24

tkellen commented on modules/formatter-jsonapi/index.js:
```diff
@@ -4,83 +4,93 @@ const getKey = require('./lib/get_key');
+var linkedResourceIndex;
+var typeName;
+
+function formatModel(output, model) {
```
can we move this into `lib/format_model.js`?

tkellen commented on modules/formatter-jsonapi/index.js:
```diff
@@ -4,83 +4,93 @@ const getKey = require('./lib/get_key');
+  var links = link(model, {
+    linkWithInclude: linkWithInclude,
+    linkWithoutInclude: linkWithoutInclude,
+    exporter: function (models, type) {
```
this will probably have to become a param to `lib/format_model.js`

tkellen commented:
:+1: other than getting that stateless method into it's own file and unit tested alone.

bobholt commented:
moved formatModel to its own file and added a unit test

this makes the first unit test in the create and update responders even less relevant - may remove those when we get full integration coverage

tkellen commented:
:tada:

tkellen commented:
Love it! merge away
